src/Illuminate/Http/Client/Request.php

```diff
@@ -149,7 +149,8 @@ public function hasFile($name, $value = null, $filename = null)
         return collect($this->data)->reject(function ($file) use ($name, $value, $filename) {
             return $file['name'] != $name ||
                 ($value && $file['contents'] != $value) ||
-                ($filename && $file['filename'] != $filename);
+                ($filename && $file['filename'] != $filename) ||
+                ! is_resource($file['contents']) && json_encode($file['contents']) !== false;
```

Comment:
As the previous checks are using OR (`||`), shouldn't these last two be wrapped in parentheses?

Also, in your description you mention the `json_encode()` is "retained", but the GIT diff doesn't show it on the previous version... 

- Where did it come from? 
- Isn't trying to encode every attachment as JSON, expecting it to fail, as a condition inside a `Collection@reject()` callback, an expensive operation?

Reply:
Regarding your first point, I see what you mean, and I will use parentheses for that. However, regarding your second point, there was a misunderstanding in my explanation. I intended to use json_encode to ensure that we only check the file attachments rather than other fields in the request body, and I didn't mean to imply that this function was already used.

This approach enhances the logic of the code by ensuring that we check whether the request contains a specific file, which is the expected functionality of this method.